src/main/state/nodes.ts

```diff
@@ -197,3 +199,31 @@ export const removeNode = (nodeId: NodeId) => {
   store.set(USER_NODES_KEY, { nodes, nodeIds: newNodeIds });
   return nodeToRemove;
 };
+
+powerMonitor.on('shutdown', () => {
+  restartNodes('shutdown');
+});
+
+powerMonitor.on('suspend', () => {
+  restartNodes('shutdown');
+});
+
+powerMonitor.on('resume', () => {
+  restartNodes('login');
+});
+
+const restartNodes = (reason: 'shutdown' | 'login') => {
+  const nodesToRestart = getNodes().filter(
+    (node) =>
+      node.status === NodeStatus.running &&
```

Comment:
can remove `node.status === NodeStatus.running`. For now, if the node is not manually stopped by the user, then we will try to keep it running